src/components/carousel/carousel.ts

```diff
@@ -0,0 +1,386 @@
+          });
+        }
+      }
+    });
+
+    this.intersectionObserverEntries.clear();
+  }
+
+  @eventOptions({ passive: true })
+  handleScroll() {
+    this.isScrolling = true;
+
+    this.handleScrollEnd();
+  }
+
+  @watch('loop', { waitUntilFirstUpdate: true })
+  handleLoopChange() {
+    const slides = this.getSlides();
+    const intersectionObserver = this.intersectionObserver;
+
+    this.getSlides({ excludeClones: false }).forEach(slide => {
+      intersectionObserver.unobserve(slide);
+
+      if (slide.hasAttribute('data-clone')) {
+        slide.remove();
+      }
+    });
+
+    if (this.loop) {
+      const lastClone = slides[slides.length - 1]?.cloneNode(true) as HTMLElement;
+      lastClone.setAttribute('data-clone', String(slides.length - 1));
+      this.prepend(lastClone);
+
+      const firstClone = slides[0]?.cloneNode(true) as HTMLElement;
+      firstClone.setAttribute('data-clone', '0');
+      this.append(firstClone);
+
+      // Because the dom has changed, restore the scroll position to the active slide
+      this.gotoSlide(this.activeSlide, 'auto');
+    }
+
+    this.getSlides({ excludeClones: false }).forEach(slide => {
+      intersectionObserver.observe(slide);
+    });
+  }
+
+  pauseAutoplay = () => {
+    clearInterval(this.autoplayTimerId);
+    this.autoplayTimerId = 0;
+  };
+
+  resumeAutoplay = () => {
+    if (this.autoplay) {
+      this.autoplayTimerId = window.setInterval(() => {
+        this.nextSlide();
+      }, this.autoplayInterval);
+    }
+  };
+
+  @watch('autoplay')
+  handleAutoplayChange() {
+    this.pauseAutoplay();
+    this.resumeAutoplay();
+  }
+
+  handlePaginationKeydown(event: KeyboardEvent) {
+    if (!['ArrowDown', 'ArrowUp', 'ArrowRight', 'ArrowLeft'].includes(event.key)) {
+      return;
+    }
+
+    event.preventDefault();
+
+    const isRtl = this.localize.dir() === 'rtl';
+    const isLtr = this.localize.dir() === 'ltr';
+
+    if ((isLtr && event.key === 'ArrowRight') || (isRtl && event.key === 'ArrowLeft') || event.key === 'ArrowDown') {
+      this.nextSlide();
+    } else if (
+      (isLtr && event.key === 'ArrowLeft') ||
+      (isRtl && event.key === 'ArrowRight') ||
+      event.key === 'ArrowUp'
+    ) {
+      this.previousSlide();
+    }
+
+    // Move the focus to the current active indicator
+    const indicator: HTMLElement = this.paginationContainer.querySelector(
+      `[role=presentation]:nth-of-type(${this.activeSlide + 1}) .carousel__paginationItem`
+    )!;
+    indicator.focus();
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+
+    const intersectionObserver = new IntersectionObserver(
+      (entries: IntersectionObserverEntry[]) => {
+        entries.forEach(entry => {
+          // Store all the entries in a map to be processed when the scroll ends
+          this.intersectionObserverEntries.set(entry.target, entry);
+        });
+      },
+      {
+        root: this.slidesContainer,
+        threshold: 0.6
```

Comment:
Yes, it determines when a slide is considered displayed. I don't know to be honest what could be the advantage of making it configurable since it is just for internal use. I'd leave it as it is for now if it's fine with you